src/main/java/de/tum/in/www1/artemis/domain/plagiarism/PlagiarismComparison.java

```diff
@@ -1,30 +1,58 @@
 package de.tum.in.www1.artemis.domain.plagiarism;
 
-import java.util.List;
+import java.util.Set;
 import java.util.stream.Collectors;
 
+import javax.persistence.CascadeType;
+import javax.persistence.CollectionTable;
+import javax.persistence.ElementCollection;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+
 import jplag.JPlagComparison;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+import de.tum.in.www1.artemis.domain.DomainObject;
 import de.tum.in.www1.artemis.domain.plagiarism.text.TextSubmissionElement;
 
 /**
  * Pair of compared student submissions whose similarity is above a certain threshold.
  */
-public class PlagiarismComparison<E extends PlagiarismSubmissionElement> {
+@Entity
+@Table(name = "plagiarism_comparison")
+public class PlagiarismComparison<E extends PlagiarismSubmissionElement> extends DomainObject {
+
+    /**
+     * The result this comparison belongs to.
+     */
+    @ManyToOne(targetEntity = PlagiarismResult.class)
+    @JsonIgnore
+    private PlagiarismResult<E> plagiarismResult;
 
     /**
-     * First submission involved in this comparison.
+     * First submission compared.
      */
+    @ManyToOne(targetEntity = PlagiarismSubmission.class, cascade = CascadeType.ALL)
```

Comment:
CascadeType.ALL does not make sense in ManyToOne, because we might get situations where multiple PlagiarismComparisons reference the same  PlagiarismSubmission and then deleting only one 
PlagiarismComparison should potentially not delete the PlagiarismSubmission which is still references in another object

Reply:
maybe it is enough to document this case better, because we only allow to delete the container object and then all child objects should be deleted as well